bip-xxxx.mediawiki

```diff
@@ -0,0 +1,358 @@
+
+The sender must be careful to only sign the inputs that were present in the original PSBT and nothing else.
+
+Note:
+* The sender should allow the payment output to be modified by the receiver (The receiver may substitute a P2WPKH payment to P2SH payment to increase privacy)
+* The sender must allow the receiver to add outputs.
+* The sender must allow the receiver to not add any input. Useful for the receiver to change the paymout output scriptPubKey type.
+* If no input have been added, the sender's wallet implementation should accept the payjoin proposal, but not mark the transaction as an actual payjoin in the user interface.
+
+Our method of checking the fee allows the receiver and the sender to batch payments in the payjoin transaction.
+It also allows the receiver to pay the fee for batching adding his own outputs.
+
+On top of those check, it is recommended, but not required for the sender to check that:
+* If the sender is making a payjoin with a change (ie, not in the [[#spare-change|spare change]] case), make sure the receiver is paying for any batched output.
+
+===Optional parameters===
+
+When the payjoin sender posts the original PSBT to the receiver, he can optionally specify the following HTTP query string parameters:
+
+* <code>v=</code>, the version number of the payjoin protocol that the sender is using. The current version is <code>1</code>.
+
+This can be used in the future so the receiver can reject a payjoin if the sender is using a version which is not supported via an error HTTP 400, <code>version-unsupported</code>.
+If not specified, the receiver will assume the sender is <code>v=1</code>.
+
+If the receiver does not support the version of the sender, they should send an error with the list of supported versions:
+<pre>
+{
+    "errorCode": "version-unsupported",
+    "supported" : [ 2, 3, 4 ],
+    "message": "The version is not supported anymore"
+}
+</pre>
+
+* <code>additionalfeeoutputindex=</code>, the preferred output from which to increase the fee for the added inputs. (default: <code>-1</code>)
+
+If the <code>additionalfeeoutputindex</code> is out of bounds or pointing to the payment output meant for the receiver, the receiver should ignore the parameter.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>maxadditionalfeecontribution=</code>, an integer defining the maximum amount in satoshis that the sender is willing to contribute towards fees for the additional inputs. <code>maxadditionalfeecontribution</code> must be ignored if set to less than zero. (default: -1)
+
+Note that if <code>maxadditionalfeecontribution</code> is too low, the sender should create a transaction with RBF disabled, as the original transaction could replace the payjoin transaction.
+Should be ignored in the [[#spare-change|spare change]] case.
+
+* <code>minfeerate=</code>, a decimal in satoshi per vbyte that the sender can use to constraint the receiver to not drop the minimum fee rate too much.
```

Comment:
I think this is useless. If the sender isn't happy with the fee rate, they should just use the original. All this really does is just introduces a code branch that rarely gets executed as the sender needs to verify the receiver honored the minfeerate anyway..

I'd also recommend "satoshis per weight" as a better unit, as it's the two sort of fundamental base units in bitcoin. Also if we're going with a minfeerate as a decimal (which makes sense) it's good to specify a rounding policy (i.e. floor, round, or ceil). I would suggest:

"minfeerate, a floating point number representing satoshis per weight, such that the transaction pays `>= ROUND(minfeerate * transactionWeight)`